pip/wheel.py

```diff
@@ -37,6 +37,12 @@
 from pip.utils.temp_dir import TempDirectory
 from pip.utils.ui import open_spinner
 
+try:
+    from concurrent.futures.process import ProcessPoolExecutor
```

Comment:
What if the backport only works on Python 2.7 and the core library only works on Python 3.4 and above? Then developers will need to spend additional time to update the vendored version.

> OTOH, If the user install a release of concurrent.futures that pip has not been tested with and pip would not work - it is pip's fault and if you think about it, it'll be impossible to use pip if it has such a try-except to recover from a situation.

Ordinarily I would agree, but concurrent.futures is a core python library. It's included in the Python distribution. Using this logic, pip would have to account for every case where users patch core python functions such as shutil.copy.